src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -861,6 +862,22 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
             return Succeeded;
         }
 
+        private void ReportRefFilenameMismatch(string finalPeFilePath, string finalRefPeFilePath, DiagnosticBag diagnosticBag)
+        {
+            if (finalPeFilePath == null || finalRefPeFilePath == null)
+            {
+                return;
+            }
+
+            var primaryOutput = Path.GetFileName(finalPeFilePath);
+            var secondaryOutput = Path.GetFileName(finalRefPeFilePath);
+            if (!primaryOutput.Equals(secondaryOutput, StringComparison.OrdinalIgnoreCase))
```

Comment:
Good question. I'm also not sure what is the proper comparison for filenames (case sensitive or insensitive?).

So, I also got feedback from Windows contact they always name their ref assembly a different name... I'm rolling back this commit and I'll follow up with Jared and Neal as a follow-up bug.
Thanks